runtime/drivers/admin/admin.go

```diff
@@ -110,11 +116,17 @@ type Handle struct {
 	cloned               bool
 	repoPath             string
 	projPath             string
-	gitURL               string
-	gitURLExpiresOn      time.Time
 	virtualNextPageToken string
 	virtualStashPath     string
 	ignorePaths          []string
+
+	// git related fields
+	// These will not be set if downloadURL is set
+	gitURL          string
+	gitURLExpiresOn time.Time
+
+	// downloadURL is set when using one-time uploads
+	downloadURL string
```

Comment:
What if someone migrates from upload deploys to Github deploys? I'm not sure if the right handling is in place to clear/reset state here. (Or maybe it's handled by some upstream invalidation logic?)

Reply:
Yes there is no handling right now. I planned to add it in follow up PRs.

Reply:
Here are possible high level approaches for handling that. I can add that handling in this PR itself if you prefer it that way.

1. Only allow it from UI and orchestrate work flow from UI. 
2. In all cases detect if a user has an existing project and if user choose to connect project to `Github` or if a git remote has been set offline, migrate the project from `one time upload` to `git` based. 
3. Introduce a separate command to connect existing project to github.

2 is preferable but adds complexity to the implementation in my opinion.

May be we can introduce this capability if customers request for it ?

Reply:
I'm not talking about adding support for the migration – agree that we can do that in a follow up PR. What I mean is that at least this file should gracefully handle that transition (e.g. can just set the download-related fields to `nil` and clone from scratch).

Reply:
I didn't handle it previously because that migration is not possible. 
Evaluating it further we don't need to do anything here. Whenever that migration happens we will do a `EditInstance` with a `nonce` set to a new value which will reopen the admin handle clearing the previous state.

Reply:
Perfect!